test/gtest/reducecalculation.hpp

```diff
@@ -231,39 +267,53 @@ struct ReduceCalculationTest : public ::testing::TestWithParam<ReduceCalculation
                                                   ws_sizeInBytes,
                                                   input.desc,
                                                   input_dev.get(),
-                                                  output.desc,
+                                                  isLogicalCalculation ? logical_output.desc
+                                                                       : output.desc,
                                                   output_dev.get(),
                                                   nanPropagation,
                                                   dim,
                                                   reduceCalculationOp);
 
-        EXPECT_EQ(status, miopenStatusSuccess);
+        ASSERT_EQ(status, miopenStatusSuccess);
 
-        output.data = handle.Read<T>(output_dev, output.data.size());
+        if(isLogicalCalculation)
+        {
+            logical_output.data = handle.Read<uint8_t>(output_dev, logical_output.data.size());
+        }
+        else
+        {
+            output.data = handle.Read<T>(output_dev, output.data.size());
+        }
     }
 
     void Verify()
     {
-        // Computation error of fp16 is ~2^13 (=8192) bigger than
-        // the one of fp32 because mantissa is shorter by 13 bits.
-        auto threshold = std::is_same<T, float>::value ? 1.5e-5 : 8.2e-2;
-
-        // bf16 mantissa has 7 bits, by 3 bits shorter than fp16.
-        if(std::is_same<T, bfloat16>::value)
-            threshold *= 80.0;
-        auto error = miopen::rms_range(ref_output, output);
-
-        EXPECT_TRUE(miopen::range_distance(ref_output) == miopen::range_distance(output));
-        EXPECT_TRUE(error < threshold * 10) << "Error output beyond tolerance Error:" << error
-                                            << ",  Thresholdx10: " << threshold * 10;
+        if(isLogicalCalculation)
+        {
+            bool is_equal = logical_ref_output.data == logical_output.data;
+
+            ASSERT_TRUE(is_equal) << "Logical calculation failed";
+        }
+        else
+        {
+            double threshold = std::numeric_limits<T>::epsilon();
```

Comment:
should use `GetTolerance()` at here instead of directly specifying it

Reply:
solved